csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs

```diff
@@ -397,6 +397,55 @@ private void TestPreTrainedModelsOpset7And8()
             } //opset
         }
 
+        [Fact]
+        private void TestOverridableInitializerMetadata()
+        {
+            string modelPath = Path.Combine(Directory.GetCurrentDirectory(), "overridable_initializer.onnx");
+            using (var session = new InferenceSession(modelPath))
+            {
+                Assert.Equal(2, session.InputMetadata.Count);
+                Assert.True(session.InputMetadata.ContainsKey("Label"));
+                Assert.True(session.InputMetadata.ContainsKey("F2"));
+
+                Assert.Equal(1, session.OverridableInitializerMetadata.Count);
+                Assert.True(session.OverridableInitializerMetadata.ContainsKey("F1"));
+                Assert.True(session.OverridableInitializerMetadata["F1"].IsTensor);
+                Assert.Equal(typeof(float), session.OverridableInitializerMetadata["F1"].ElementType);
+                Assert.Equal(2, session.OverridableInitializerMetadata["F1"].Dimensions.Length);
+                Assert.Equal(1, session.OverridableInitializerMetadata["F1"].Dimensions[0]);
+                Assert.Equal(1, session.OverridableInitializerMetadata["F1"].Dimensions[1]);
+
```

Comment:
Can the shape of the OverridableInitializer be overridden? If so, we could include a test to see if an initializer of shape of (1,1) can be overridden with values of shape (n,n).

Reply:
Does not look like the initializer shape is overridable, at least from the C-API. Let me know if there is any specific example/case of that.